### ISSUE TITLE
Mozilla Study Groups in Brasil

### ISSUE THREAD

**Issue by katieannemills** [mozillascience/studyGroup#36]
@rodrigopadula and I were just talking about how great it would be if we could get a series of Study Groups started up around Brasil. But, we need a plan to move this forward effectively. Some rough steps we talked about:
1. Get this repo translated into Portuguese on a Portuguese branch (help wanted!). Translation of other assets like the Handbook can come later, but this repo is key.
2. Start a few groups around Brasil. I know @pcjunger has one coming for UFRJ, and @rodrigopadula is looking into ICUFF. One or two more around the country would be a great start.
3. Once there are two or three groups started, we should start a monthly google hangout in Portuguese, lead by the local community. We do one in english so far (which everyone is welcome to join in on), but one organized regionally would be a great addition.

This is a pretty simple plan so far - any ideas or details I'm missing? Please feel free to forward / cc anyone else who might like to help out with this!

**Comment by pcjunger** [comment]
That sounds like a great starting point @BillMills and @rodrigopadula. I'm currently approaching the end of my Masters, thus time is limiting, but I will be glad to spend all time I have left to help out with this plan. 

I've contacted a few colleagues from other Universities who might be interested in founding more R study groups around Brazil. @paternogbc @ewaldoj @LucianoFSgarbi would you guys be interested in starting up a Mozilla Science Study Group at UFRN and UFG, respectively? It couldn't be simpler than it actually is. Let us know in case you have any doubts. The more we are, the quicker we can step forward!

**Comment by rgaiacs** [comment]
@ctruciosm, @lbenitesanchez, @ocefpaf and @fernandomayer Do you have interest on this?

**Comment by paternogbc** [comment]
Hi all,

Could you please provide more information about this idea? 
What is the main aim of these study groups?

Cheers,
G

**Comment by lfsgarbi** [comment]
What is the aim of this study group?

2015-10-06 8:48 GMT-03:00 Gustavo Paterno notifications@github.com:

> Hi all,
> 
> Could you please provide more information about this idea?
> What is the main aim of these study groups?
> 
> Cheers,
> G
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/mozillascience/studyGroup/issues/36#issuecomment-145833482
> .

## 

Luciano F. Sgarbi
Currículo Lattes
http://buscatextual.cnpq.br/buscatextual/visualizacv.do?id=K4299200T7
Laboratório de Ecologia Teórica e Síntese
Departamento de Ecologia, Instituto de Ciências Biológicas V.
Universidade Federal de Goiás, campus II, Goiânia-GO, Brasil

**Comment by pcjunger** [comment]
Hi @paternogbc and @LucianoFSgarbi, sorry for including you without providing further information. This idea is all about open science. You can find the general idea in the readme.md file in this repository: https://github.com/mozillascience/studyGroup

This Study Group repo is part of the Mozilla Science Lab project: https://www.mozillascience.org

I've just created a group for ecologist at UFRJ, which you can access here: https://github.com/pcjunger/studyGroup
It would be awesome building a great network in Brasil.

Anyway, that's all very new for me. Perhaps @BillMills can give you more details and properly answer your questions.

**Comment by paternogbc** [comment]
Hi @pcjunger, thanks for the update! ;)

I will have a look, but the idea sounds very interesting. 
I am very kin to open science and reproducible research. 

I can start a group in my university, just need to check how the group works and what are the activities that we should organize.

All the best,
G

**Comment by rgaiacs** [comment]
If people want to talk in Portuguese let me know. =)

**Comment by paternogbc** [comment]
Português então. kkk Fiquei confuso pq todo mundo escreveu em inglês. Mas vamos lá.

O primeiro passo então é traduzir esse repositório para pt, certo?
Eu posso criar o "branch" inicial e começar a tradução. Alguém pode ajudar nessa tarefa?

Abraços,
Guga

**Comment by paternogbc** [comment]
@BillMills should we create a portuguese branch or a fork from the original repo?

Thanks,
G

**Comment by pcjunger** [comment]
hahaha..continuei escrevendo em inglês pq o cara que abriu a discussão não é brasileiro, mas uma das ideias é preparar o terreno para que as discussões possam acontecer em português tbm :-) 

De qualquer forma, obrigado pelo entusiasmo! Eu posso te ajudar nessa tarefa @paternogbc, let's wait for @BillMills reply.

**Comment by rodrigopadula** [comment]
Olá pessoal, desculpem-me pela demora. Estou bem enrolado com o doutorado esses dias mas vamos lá!!!

A idéia é estimular a criação de vários grupos de estudo em universidades brasileiras para que possamos desenvolver projetos, estudos e principalmente compartilha código e conteúdo através do github.  Venho conversando com o pessoal da Mozilla há algum tempo e a idéia é fazer um intercambio entre academia e comunidade mozilla também.  Temos atualmente uma comunidade Mozilla Brasil bem forte www.mozilla.org.br e pretendemos envolver mais gente na luta por uma web cada vez mais aberta e democrática, acessível a todos!

**Comment by rodrigopadula** [comment]
Conversei com o @BillMills no último domingo e concluimos que a melhor estratégia seja criar pequenos grupos em nossas universidades, amadurecer um pouco mais a idéia e daí poderemos partir para eventos e encontros regionais/nacionais. Criar um hub local entre universdades de um mesmo estado/cidade ja seria um ótimo começo. Estou atualmente na UFF mas fiz meu mestrado na UFRJ no PPGI. Então podemos contar com apoio do pessoal de computação de ambas! Comentei com o @BillMills  que vou solicitar à mozilla a criação de uma lista de emails para que possamos trocar idéias e convidar mais gente para esse empreitada! Qualquer dúvida podem entrar em contato comigo também, participo das atividades da Mozilla no país desde 2006 ativamente.

**Comment by Ewaldoj** [comment]
Olá pessoal, boa noite. Também estou meio perdido, mas vou me ambientar no
site e pegar mais informações para discutir melhor.

@paternogbc, sou aluno de graduação da Ecologia (UFRN) e posso falar com
você na quinta feira, antes ou depois da sua apresentação do bioforum,
sobre esse assunto. Eu me apresento para você lá, :)

**Comment by paternogbc** [comment]
Opa beleza, valeu @Ewaldoj! Nos encontramos por lá então.

**Comment by katieannemills** [comment]
Hi everyone,

@paternogbc & @LucianoFSgarbi, what was said by @pcjunger is pretty much exactly correct; Study Groups are informal, community organized meetups to talk about open science, reproducible research, and the skills needed to do that. Existing Study Group activities have been:
- **Hands on demo sessions** of a tool, programming language or other technique. These are the most popular sessions, since they focus on teaching & building skills. You can see links to lists of many of the lesson notes [here](https://github.com/mozillascience/studyGroupLessons/issues).
- **Coworking sessions**, where people bring whatever they are working on at the time, and work on it while talking to each other and looking for chances to get feedback or collaborate.
- **Journal clubs**, where someone suggests a couple of papers and the group discusses them.
- **Code review sessions**, (a new thing just starting out), where someone presents some code they've written, and everyone discusses ideas for improving or learning from the example.

These are just a few ideas, but the most important thing is that **Study Group is lead by you and your community** - whatever your attendees want to talk about or work on is what you should do. Then, we're trying to encourage people to share back their lessons or ideas with other Study Groups around the world, so we can learn from each other and reuse each other's work.

@paternogbc, I would love to see a `portuguese` branch in this repo. It's important to have all the forks coming from the same repo, since that's how I keep track of who is joining the program and starting new groups. With a `portuguese` branch, it would be possible to serve the website in Portuguese in two lines:

```
branch -D gh-pages
git branch -m portuguese gh-pages
```

Actually, it may even be possible right from within github, without having to do any command line git at all. Let me know what you think, and if there are any ways I can help!

**Comment by paternogbc** [comment]
Dear @BillMills,

this sounds very nice! Thanks for putting everything together! 
I think we can engage a nice community in Brazil to participate in this project. I am not sure what is the best way to start this interaction (at the local or regional scale). But we will find a way.

Regarding the `portuguese` branch, I have no permission to push a remote branch on your repo. I am not sure if this is what you mean. Or should I create a fork, then a `portuguese` branch, then push and create a pull request? Could you please give me some guidelines? I got a bit confused regarding the workflow you suggested. **Thanks** :smiley: 

I would say we should arrange a skype meeting to exchange some ideais among the brazilians and to define our organization strategy.  We could organize ourselves from the regional to the local scale. Then, discuss some guidelines and start building local communities based on this discussions and on previous experience of people in this group. What do you think guys? @pcjunger @rgaiacs @LucianoFSgarbi @rodrigopadula 

All the best,
G

**Comment by brennovich** [comment]
@paternogbc you don't need to push directly to @BillMills' repository, instead you should fork @pcjunger's repository, develop your work under `portuguese` branch, and then create a [Pull Request](https://help.github.com/articles/using-pull-requests/) of your branch to @pcjunger's repository. After this process @pcjunger can do what @BillMills proposed :)

However, looking at this process it seems too manual and not so friendly to newcomers. Wouldn't be better to have full **I18n** support as part of `studyGroup` website project? @BillMills, I can help you guys with that if you think it's worth :)

**Comment by paternogbc** [comment]
Hi @brennovich, for me it only depends on what you prefer to keep track of new groups joining the project.
I can fork @pcjunger repo and create the portuguese branch from there. No problem ;)

All the best,
G

**Comment by katieannemills** [comment]
Hi @paternogbc & @brennovich,

I just created a new `portuguese` branch of the main repo - if you send a pull request against that branch, I will merge in your translation there. To do so, I think it will be easiest to fork this repo, make your translation on your `gh-pages` branch, and use the GitHub UI to set up the PR, like so:

![screen shot 2015-10-06 at 10 22 29 pm](https://cloud.githubusercontent.com/assets/1896943/10329633/cbc4a3f0-6c78-11e5-9049-61641c101ad0.png)

Sorry this is a bit complicated! Let me know if I can help - if you get a translation on any branch of your own repo, I think I can pull it in to my `portuguese` branch in any case - let me know any questions and thanks for your effort!

**Comment by diogoprov** [comment]
Olá! Eu também gostaria de começar um grupo de estudo. @LucianoFSgarbi posso te ajudar se quiser.

**Comment by paternogbc** [comment]
@BillMills thanks!

That sounds much better! 

I have created the fork [https://github.com/paternogbc/studyGroup], so we can work on the translation there. 

Cheers.

**Comment by diogoprov** [comment]
@paternogbc I can help with the translation as well!

**Comment by paternogbc** [comment]
Nice @diogoprov! 

Vamos trabalhar juntos então nesse [fork](https://github.com/paternogbc/studyGroup). 
Talvez seja melhor se trabalharmos em arquivos diferentes.
Pelo que eu vi tem o arquivo [README.md](https://github.com/paternogbc/studyGroup/blob/gh-pages/README.md) e o [codeOfConduct.md](https://github.com/paternogbc/studyGroup/blob/gh-pages/codeOfConduct.md)

Não sei se tem mais arquivos para serem traduzidos, mas certamente esses são os principais.
Qual vc prefere trabalhar?

abs,
G

**Comment by ocefpaf** [comment]
@rgaiacs valeu pelo toque.

@BillMills I know some that people at [UFBA](https://en.wikipedia.org/wiki/Federal_University_of_Bahia) are interested in creating a study group.  I will get them involved.

Also, I can help with the translation and/or as a reviewer. @paternogbc feel free to ping me from your fork (to avoid the noise in this thread) if you need some review.

**Comment by paternogbc** [comment]
@ocefpaf,
Thanks for the support! ;)

Lets move the discussion to the fork then.

All the best,
G

**Comment by diogoprov** [comment]
@paternogbc eu posso ficar com o codeOfConduct.md . Não tem um issue aberto no fork, então não sei como fazer pra coordenar o trabalho lá

**Comment by paternogbc** [comment]
@diogoprov Não posso criar issues nesse fork, mas você pode simplesmente editar o arquivo no meu fork. Quando terminar faça um commit e depois um pull request. ;) 

Quando terminarmos as traduções a gente da um ping no pessoal para eles revisarem.

Se quiser podemos conversar por e-mail também para evitar mais mensagens por aqui.
Abs,
Guga

**Comment by rgaiacs** [comment]
Great discussion.

@BillMills I suggest to have https://github.com/mozillascience/studyGroup-pt or https://github.com/mozillascience/studyGroup-portuguese or https://github.com/mozillascience/GrupoDeEstudo because it will be more easy to find by everyone. For example, someone looks at https://github.com/mozillascience/ will discovery that there is something in Portuguese without difficult and the last suggestion also make easy to someone find about it when Googling or searching on GitHub. This approach also has two advantages:
1. we will not bother everyone with the translation.
2. will be easy to Git novices to contribute back.

I think that everyone on this conversation right now is a Git Jedi and can handle remote repositories, merging https://github.com/mozillascience/studyGroup/ into the translated version, ... without difficult.

> Comentei com o @BillMills que vou solicitar à mozilla a criação de uma lista de emails para que possamos trocar idéias e convidar mais gente para esse empreitada!

Another suggestion is using GitHub issues as a mailing list because will make easy to keep track of the conversation. Mixing mailing list, forum, wiki, bug track just create confusion.

**Comment by rodrigopadula** [comment]
Pessoal, o @BillMills  ja migrou, só traduzir o conteúdo de https://github.com/mozillascience/studyGroup/tree/portuguese

**Comment by paternogbc** [comment]
Ok.. just created a fork from this branch then. @diogoprov, I am moving the repo, lets work [here](https://github.com/paternogbc/studyGroup).

I think the mailing list is a great idea. Let us know when you have that ready.

G

**Comment by rodrigopadula** [comment]
Galera, não precisa criar um novo fork, só trabalhar direto na arvore em português, só traduzir direto lá. Devemos trabalhar diretamente aqui https://github.com/mozillascience/studyGroup/tree/portuguese

**Comment by paternogbc** [comment]
We are going crazy :tada: 

@rodrigopadula Eu não sei esse é o melhor jeito pq toda mudança que fizermos terá que ser aprovada individualmente pelo dono do repositório. Eu acho que a sugestão do Bill é mais prática. Trabalhamos no fork e quando terminarmos a tradução criamos um pull request para o branch portuguese que ele já criou.

Eu acho que isso facilita a vida do dono do repositório e podemos trabalhar em arquivos locais sem ter que estar no Github.
Se vc achar que essa não é a melhor opção, eu sigo suas recomendações. =))
Veja o que acha.

Grande abraço,
G

**Comment by rodrigopadula** [comment]
Qualquer coisa ele pode mudar permissões nessa arvore em pt nos dando acesso direto. Assim ja fazemos todos ali mesmo, o que acham? Ficar mais facil pra manter e gerir futuras atualizações.

**Comment by paternogbc** [comment]
Se ele der autorização ta tranquilo.

abs
ps. Eu já comecei um parágrafo de [teste](https://github.com/paternogbc/studyGroup/commit/af3af8d1b7906e18414948772bbfc903bef7f99d), se mudarmos para o branch do Bill ai eu levo essas mudança.

**Comment by katieannemills** [comment]
Thanks everyone for such a great conversation! Couple of ideas:
- @rgaiacs, I think your idea of having a Portuguese-first repo makes a lot of sense. Mozilla can give this community management rights to that repo so the Brasilian team can commit directly to it. I originally suggested a branch of the central repo, because I have a lot of infrastructure monitoring that repo (webhooks email me whenever someone forks it, so I can get in touch with them) - but we can set this up for the Portuguese repo too, it is not so hard. I'm happy with whatever the community will find most convenient - but in any case, we just need to get a translation done at all, then we can decide where it should live.
- if people have the time, an extra goal could be to look into translating the [Study Group Handbook](http://mozillascience.github.io/studyGroupHandbook/index.html) - this is the complete manual for all the collected ideas and strategies on how to run a Study Group. This is a _much_ bigger translation project, but one paragraph a day and we'll be done in no time.

Thanks again, everyone! I would be so happy to bring these meetups to Brasil, but it can't be done without your help.

**Comment by rgaiacs** [comment]
Please ping me to review the translation.

**Comment by paternogbc** [comment]
Ok @rgaiacs ... when we finish translating the files we will ping you to review ;)

Thanks for helping.
G

**Comment by katieannemills** [comment]
Hey everyone, how are things going? Check out @gabano's translation [here](http://gabano.github.io/studyGroup/). It would be good to pull everyone's hard work together into that upstream branch or fork we talked about; once everyone here agrees on a first version, we can set that up. Let me know!

**Comment by katieannemills** [comment]
Also, as I just mentioned over on @gabano's repo, we're going to be having our monthly Study Group Leader's teleconference on October 30 - this would be a great chance to ask questions and meet other Study Group leaders worldwide. If you want to participate, get on [this mailing list](https://mail.mozilla.org/listinfo/msl-study-groups) - I'll be sending contact info out there before the date.

Long term, as I think I mentioned further up this thread, it'd be great to do a Portuguese version of these leaders' calls for all the groups operating in Brasil; that'll be up to you to self-organize, but for now you can jump on the English call just to meet some people. Keep me posted on this work, hopefully I see some of you on the 30th!

**Comment by paternogbc** [comment]
Very nice @BillMills .... everyone can check and review the "CodeOfConduct" translation [here](https://github.com/paternogbc/studyGroup/blob/gh-pages/codeOfConduct.md). Thanks to @diogoprov.

I am working on the [README.md](https://github.com/paternogbc/studyGroup/blob/gh-pages/README.md) file.

Is that correct @BillMills ? Or we are translating the wrong files?

I was not sure what to translate. But at least we have two files ready for review, the one you sent and  CodeofConduct.
The teleconference sounds very exciting ;)

All the best,
Guga

**Comment by gabano** [comment]
Hi everyone, 
I was in a R Workshop at UFRJ and Melissa (the teacher) invited me to lead a study group for our department. Since then I've been working on a fork and trying to translate it. Also, we had our first meeting today and worked on Git and GitHub. 

I didn't expect to see that much of people working on this in Brasil, and it's a nice surprise that you guys are working on _studyGroup_ translation. 

The translation I did isn't free of errors and many things were done based only on personal taste, because, again, I didn't expect this much of people working on this. Also, I dropped some lines of the code, mainly the calendar because I wasn't going to use it, so be careful with that.

Let me know when you decide on which branch you are going to work and if I can help. 

joao.

**Comment by pcjunger** [comment]
Hi @gabano, I've met Melissa. She is from the same UBC research group of the guys who introduced the Mozilla Study Group to us at PPGE - UFRJ. :-)
I've noticed you have already translated the whole website. That's very nice! I've also noticed you started translating the README file. That's is also nice, but @paternogbc has already started working on it and we could decide a single file to work on and avoid double-working on the same file. 

Pelo jeito os grupos estão surgindo e o grupo brasileiro está crescendo rapidamente. Excelente! Ao mesmo tempo precisamos nos organizar melhor e evitar que trabalhemos duplamente no mesmo arquivo. Confesso que estou um pouco confuso, talvez por ser novo por aqui. Bom, que tal centralizarmos aqui quais serão os arquivos padrão, minha sugestão é:
- "README.md": todos revisarmos o arquivo README que o @paternogbc está traduzindo;
  - "CodeOfConduct": Já foi integralmente traduzido pelo @diogoprov. Quem quiser pode fazer revisões no link que @paternogbc mandou acima.
  - Website: @gabano você poderia compartilhar aqui o link do arquivo que você modificou para traduzir a página completa? Não sei qual foi.

É isso, a ideia da reunião também parece bem interessante. Valeu gente!

Abs,
Pedro

**Comment by paternogbc** [comment]
Hi all,

the **README** file is nearly read. If anyone wants to review or help to finish the translations, please change this [file](https://github.com/paternogbc/studyGroup/blob/gh-pages/README.md).

All the best,
G

**Comment by diogoprov** [comment]
Hi guys,
I'd really appreciate if someone could take a look and review my translation. @paternogbc I'll work from now on to finish the translation of the README file. 

Best,
Diogo

**Comment by katieannemills** [comment]
@paternogbc yes, that's perfect! Once those are translated, we will be ready to offer them as a complete translation of the website. I know we talked about doing this as a branch in the original repo, but @rgaiacs has convinced me - I think it might be easier if we make a whole new repo to house this translation. That way, users can just fork the translation they want without having to swap branches around, and I can make everyone here a maintainer of the Portuguese translation so you can make changes and updates without my oversight. I'll make sure this is ok with Mozilla (since it's their github org) and set something up!

**Comment by paternogbc** [comment]
@BillMills thanks for your reply. I agree that a separate repo is the best way to do it.
We have decided to use a similar strategy in the portuguese translations of swirl courses for R. [swirl-BR](https://github.com/paternogbc/cursos_swirl-Br) where we started a fresh repo to avoid problems.

I believe we should have all translations ready by next week. Actually, we are at the revision phase. ;)
After we finish, we then decide who is going to hold the base repo and all study groups can fork this and start their on websites.

Who wants to build the base brazilian repo? You will need to compile all translations and push to this repo, then all groups can fork that. =)

@BillMills please let us know if Mozilla agrees with this approach. 

All the best,
G

**Comment by katieannemills** [comment]
@paternogbc - sounds good! I just asked Moz if we could host it under [their org](https://github.com/mozillascience) and give everyone here write access; I suspect there will be no problems with that, but just wanted to ask them before charging ahead. I'll let you know!

**Comment by paternogbc** [comment]
Ohhh this is even better! 

@BillMills thanks a lot for putting everyone together!
G

**Comment by gabano** [comment]
@pcjunger   eu traduzi o conteúdo da pasta _includes

---

Everyone,

I've finished translating the contents of  /_includes and set up a `portuguese` branch for it since `gh-pages` is filled with our own studyGroup info. 

Let me know for whom to push it and make the pull request. 

joao.

**Comment by abbycabs** [comment]
Hey all! 

On advice from @alicoding, the easiest thing for translation might be to to add the translation as another directory on the gh-pages branch.  `/studyGroup/pt-br` and `/studyGroup/en`

**Comment by alicoding** [comment]
I would suggest to put the locale first e.g. `/en/studyGroup/`

**Comment by abbycabs** [comment]
@alicoding - that makes more sense... :)

**Comment by rgaiacs** [comment]
> On advice from @alicoding, the easiest thing for translation might be to to add the translation as another directory on the gh-pages branch. `/studyGroup/pt-br` and `/studyGroup/en`

Having two repositories is good to track the changes at the original document with Git. Just my two cents.

**Comment by katieannemills** [comment]
I'm open minded to either option, but the two-repo solution lets us make the folks on this thread maintainers of their translation - though I suppose we could just assign PRs on the same repo to different maintainers. Any strong feelings? Delighted to have this community as comaintainers!

**Comment by paternogbc** [comment]
Hi @BillMills and all,

Thanks a lot for all this work.
Could please give us some more information about the next steps?

I believe we now have everything translated, although we still need to integrate all files into a single repo.

Next week I will be in a rush, but after that I can continue to work on the fork.

All the best,
Guga

**Comment by katieannemills** [comment]
Thanks, everyone - a few things to sort out so we can finish up:

@alicoding @acabunoc  do you have an example of your directories suggestion working with jekyll on gh-pages? I need some help understanding how to do that in a non-convoluted way. Problem is, I think that github essentially does `jekyll serve` from the root directory of the repo, so this isn't just as simple as moving everything into an `/en` directory and having another copy for `/pt-br`. The only way I can see to do this is to have a separate entry in `_posts` for each language, permalinked to `en` or `pt-br` as appropriate to get the route you want (though studyGroup has to come first, since it's the repo name), and then define a different `layout` for each of them that include the appropriate translations. At which point I'm not sure what would even happen to the routes of the actual event postings people are supposed to put in `_posts` - I guess those would just lose the language part of the route, which isn't explicitly broken, but is pretty weird.

This seems awful complicated - unless you have a way to simplify things?

@paternogbc working on it! As soon as we sort out how we want to host multiple translations.

**Comment by brennovich** [comment]
> I need some help understanding how to do that in a non-convoluted way. Problem is, I think that github essentially does jekyll serve from the root directory of the repo

@BillMills actually Github runs `jekyll build` everytime you update the `gh-pages` branch through **git hooks**, and then serve the generated static files.

> This seems awful complicated - unless you have a way to simplify things?

When you guys was discussing this approach of having a `pt-br`branch I tried to explain that this approach wasn't that simple. https://github.com/mozillascience/studyGroup/issues/36#issuecomment-146064322

AFAIK the original `studyGroup`project should contain all the core of the site, maintaing Jekyll logic, styles, base configuration, translations, and all other desired features. Forked `studyGroup` projects should only defined their own configuration and posts. This way we can always add new features to the main project and update forked ones:

``` sh
# From a forked project

# Add mainproject as a remote, this way we can pull latests changes
git remote add mozillascience git@github.com:mozillascience/studyGroup.git

# Fetch changes
git fetch mozillascience

# Apply changes to our fork
git merge mozillascience master

# Push updates to our fork
git push origin master
```

So, the question is, how to support multiple languages? Well, every multilingual project out there maintains their translations in **locale files** that can be yaml files (just like our config.yml). You guys can take a look in an example [here](https://github.com/teambox/teambox/tree/dev/config/locales). 

Once we have our translation in locale files like that we can specify our current locale through a configuration key or a URI param, and use translation keys into our layouts/templates https://github.com/teambox/teambox/blob/dev/app/views/people/index.haml#L21-L23

I'll try to add open a Pull Request with basic internationalization support in the next few days, it will be easier to understand ;)

**Comment by brennovich** [comment]
Just found a cool blog post https://developmentseed.org/blog/multilingual-jekyll-sites/ about it

**Comment by katieannemills** [comment]
@brennovich this is fantastic, thanks for looking into this! I definitely agree that using frontmatter (like in the blog post you linked) and locale files makes the most sense, now that you've walked us through it. As long as it is simple for a new group to fork the repo and set their website up in their language of choice (like by setting a single language variable in `_config.yml`, or by having everything behind an appropriate route, exactly as you described), then I think this is an excellent solution.

Thanks for pulling this together, and I look forward to the PR - let me know how I can help!

**Comment by brennovich** [comment]
Awesome :)

I'll let you guys know when I get something working ;)

**Comment by paternogbc** [comment]
Thanks @BillMills @rgaiacs @brennovich for organizing everything! ;)

**Comment by katieannemills** [comment]
Hi folks,

Just a reminder that we're going to have our next Study Group Leader's call on **Friday, October 30 at 3 PM Rio time** - if you're thinking of starting a Study Group, this is your chance to ask questions of your fellow leaders, and see what everyone else has been doing. We'll be connecting via google hangout; the link will be posted shortly before the meeting on [this etherpad](https://etherpad.wikimedia.org/p/studygroup-leaders-october). You are all welcome to add questions or topics to the agenda, or just ask in person during the meeting - hope to see you all there!

**Comment by rgaiacs** [comment]
@BillMills Thanks for the invitation. Unfortunately I can't make it. =(

**Comment by paternogbc** [comment]
@BillMills ... thanks for this invitation! Unfortunately I will be on a trip during this time! ;( 
I will try to catch up with the brazilians that manage to attend!

**Comment by katieannemills** [comment]
Hi folks - okay, I just merged in the infrastructure to support localization yml files, like @brennovich suggested. How to add a translation:
- Put your translated text for the website in a file like `_data/en.yml`, with an appropriate language prefix.
- in the top level `_config.yml`, set the `language` variable to the same prefix you chose above.

And that's it! Your website should use the text in your new yml file. Note that the headlines are still in the top level `_config.yml`, since people change those all the time.

For the README, please add a translated README at the same level with an appropriate language tag in the name, like `README-pt.md`. Then, at the very top of the main `README.md`, put a line in Portuguese saying '[Read this in Portuguese](link to README-pt.md)', linking to your README.

Let me know how this works for everyone, and huge huge thanks to everyone again for forming this translation and helping figure out how best to serve it. Once the text is merged in and it works for everyone, we can get started with Study Groups across Brasil.

**Comment by rodrigopadula** [comment]
@BillMills https://developer.mozilla.org/en-US/docs/Implementing_Pontoon_Mozilla  can be a good way to have things localized more quickly with support from our Mozillians fellows, what do you think?

A lot of Mozilla's sites/projects was migrated to Pontoon recently.

**Comment by katieannemills** [comment]
@rodrigopadula nice find! But will this infrastructure plug into a github-hosted page? It seems like a lot of engineering & dependencies compared to the local yml files @brennovich suggested - best to keep things as simple as possible!

**Comment by rodrigopadula** [comment]
@BillMills probably for the study groups it dont make sense now, but for https://www.mozillascience.org/ can be a good idea to have it localized in all possible languages through Pontoon.

**Comment by katieannemills** [comment]
@rodrigopadula totally - @acabunoc, pinging you to put pontoon localization for https://www.mozillascience.org/ on your radar.

Meanwhile, looking forward to `pt.yml` here - let me know if I can help!

**Comment by katieannemills** [comment]
Hey everyone - how's it going with that .pt yml file? Let me know if there's any way I can help out!

**Comment by pcjunger** [comment]
Hi everyone, sorry for taking too long to reply. I've tried to follow your last steps but to be honest I'm a bit lost at moment. If I got it right we only need to put all translated files into the the new infrastructure built by @BillMills based on what @brennovich suggested. Right? @gabano @paternogbc @diogoprov  Shall we step forward? Let me know how I could be of any help!

**Comment by gabano** [comment]
@pcjunger and everyone,   I made an effort to translate the en.yml file this morning. I added a pt.yml to [a fork](https://github.com/gernupem/studyGroup) from @BillMills.  I didn't use my own repo because I'm already running a StudyGroup there.

Please, check it out to see if everything is fine and for proofreading  :sweat_smile: . Just for the record, I didn't translate anything of the _config.yml file.

Joao.

**Comment by paternogbc** [comment]
Hi @BillMills @gabano @pcjunger and all,

I got a bit confused about how to proceed. I am ready to help. 
@gabano we could focused on the link you sent and check the pt.yml file first.
One question, what or how should we work with the [README.md](https://github.com/paternogbc/studyGroup/blob/gh-pages/README.md) and [codeOfConduct](https://github.com/paternogbc/studyGroup/blob/gh-pages/codeOfConduct.md) files that we translated and reviewed?

All the best,
Guga

**Comment by paternogbc** [comment]
I have checked the [pt.yml](https://github.com/gernupem/studyGroup/blob/gh-pages/_data/pt.yml) file from @gabano. It looks good to me! =)
@BillMills could you please suggest the next steps? 
Should I add the README and codeOfConduct files to this repo? Or should we work on other files?
This will be the definitive repo for the portuguese translations?

Sorry for so many questions!

**Comment by katieannemills** [comment]
Hi everyone,

Sorry for the slow reply, I've been traveling and catching up with everyone for the past couple days. Anyway, the pt.yml looks great, thanks for pulling that together! Send me a PR and we'll get that merged in right away.

@paternogbc for the README and code of conduct, it's a bit easier since they're not part of the jekyll formatting stuff; just add `README-pt.md` and `codeOfConduct-pt.md` to the top level of the repo, and I will link them appropriately. As for the definitive repo, that'll be this one right here once @gabano sends me that PR with `pt.yml`. Thanks again everyone, I am really excited for this to land!

**Comment by pcjunger** [comment]
Thanks @gabano! I also think the pt.yml is very well translated!
Good to see we're nearly ready with these initial steps. Looking forward to seeing mozilla study groups flourishing around Brasil!

**Comment by paternogbc** [comment]
Hi @BillMills ... thats sounds great!
Just created a PR to add both README-pt.md and codeOfconduc-pt.md to the repo.
Let me know if you need any thing else!

Also looking forward to start organizing the brazilian Mozila study groups.

**Comment by katieannemills** [comment]
Alright, that does it - I just merged and linked the translation files, so we are good to go for launching these sites in Portuguese! To recap, just fork this repo and in the in the top level `_config.yml`, set the `language` variable to `pt`, and everything should appear in Portuguese. Let me know if this works, and a HUGE thank-you to everyone who participated here!

Going forward, after getting websites set up as above, the first step for getting a Study Group started is to have a few casual meetings with people at your institution, to find people who would like to co-organize with you; the more people who help out running a Study Group, the more successful they are. I usually start with an event at the local pub to get people out to chat :)
